src/core/Akka/Serialization/Serialization.cs

```diff
@@ -91,7 +92,8 @@ public Serialization(ExtendedActorSystem system)
                 var serializer = serializerConfig != null
                     ? (Serializer)Activator.CreateInstance(serializerType, system, serializerConfig)
                     : (Serializer)Activator.CreateInstance(serializerType, system);
-                AddSerializer(serializer);
+
+                AddSerializer(kvp.Key, serializer);
                 namedSerializers.Add(kvp.Key, serializer);
```

Comment:
@Danthar so aren't we doing the same thing with ` _serializerIdByName` and the `namedSerializers` local here? Why not just use that?

Reply:
Because we already have a dictionary mapping id => serializer instance. I could have promoted the `namedSerializers` collection to a private member var. But that would mean we would store a reference to the same instance twice.

Not sure what my rationale was when i was writing this, but i wanted to avoid that. Hence the separate map `_serializerIdByName`